gobblin-service/src/main/java/org/apache/gobblin/service/modules/orchestration/FlowTriggerHandler.java

```diff
@@ -170,28 +174,80 @@ private void scheduleReminderForEvent(Properties jobProps, MultiActiveLeaseArbit
-    // Ensure we save the event timestamp that we're setting reminder for to have for debugging purposes
-    // in addition to the event we want to initiate
-    jobProps.setProperty(ConfigurationKeys.SCHEDULER_EVENT_TO_REVISIT_TIMESTAMP_MILLIS_KEY,
-        String.valueOf(status.getEventTimeMillis()));
-    jobProps.setProperty(ConfigurationKeys.SCHEDULER_EVENT_TO_TRIGGER_TIMESTAMP_MILLIS_KEY,
-        String.valueOf(originalEventTimeMillis));
-    JobKey key = new JobKey(flowAction.getFlowName(), flowAction.getFlowGroup());
-    // Create a new trigger for the flow in job scheduler that is set to fire at the minimum reminder wait time calculated
-    Trigger trigger = JobScheduler.createTriggerForJob(key, jobProps,
-        Optional.of(createSuffixForJobTrigger(status.getEventTimeMillis())));
+    JobKey origJobKey = new JobKey(jobProps.getProperty(ConfigurationKeys.JOB_NAME_KEY, "<<no job name>>"),
+        jobProps.getProperty(ConfigurationKeys.JOB_GROUP_KEY, "<<no job group>>"));
+    // Triggers:job have an N:1 relationship but the job properties must remain constant between both, which does not
+    // allow us to keep track of additional properties for reminder events. By reusing the same job key, we either
+    // encounter an exception that the job already exists and cannot add it to the scheduler or have to overwrite the
+    // original job properties with the reminder event schedule. Thus, we differentiate the job and trigger key from the
+    // original event.
+    JobKey newJobKey = new JobKey(origJobKey.getName() + createSuffixForJobTrigger(status.getEventTimeMillis()),
+        origJobKey.getGroup());
     try {
-      log.info("Flow Trigger Handler - [{}, eventTimestamp: {}] -  attempting to schedule reminder for event {} in {} millis",
+      if (!this.schedulerService.getScheduler().checkExists(origJobKey)) {
+        log.warn("Attempting to set a reminder for a job that does not exist in the scheduler. Key: {}", origJobKey);
+        this.jobDoesNotExistInSchedulerCount.inc();
+        return;
+      }
+      JobDetailImpl jobDetail = (JobDetailImpl) updatePropsInJobDetail(origJobKey, cronExpression,
+          status.getEventTimeMillis(), originalEventTimeMillis);
+      // Create a new trigger that is set to fire at the minimum reminder wait time calculated
+      Trigger trigger = JobScheduler.createTriggerForJob(newJobKey,
+          (Properties) jobDetail.getJobDataMap().get(GobblinServiceJobScheduler.PROPERTIES_KEY), Optional.absent());
+      log.info("Flow Trigger Handler - [{}, eventTimestamp: {}] -  attempting to schedule reminder for event {} in {} "
+              + "millis", flowAction, originalEventTimeMillis, status.getEventTimeMillis(), trigger.getNextFireTime());
+      this.schedulerService.getScheduler().scheduleJob(jobDetail, trigger);
+      log.info("Flow Trigger Handler - [{}, eventTimestamp: {}] - SCHEDULED REMINDER for event {} in {} millis",
```

Comment:
do we really need to log the same info in a before-calling and after-the-call variant?

Reply:
Leaving it in for now since I'm having trouble debugging this and verifying that it is working. I added a TODO to remove it later.